cmd/earthly/subcmd/build_cmd.go

```diff
@@ -603,11 +603,16 @@ func (a *Build) ActionBuildImp(cliCtx *cli.Context, flagArgs, nonFlagArgs []stri
 	buildOpts.MainTargetDetailsFunc = func(d earthfile2llb.TargetDetails) error {
 		a.cli.Console().VerbosePrintf("Logbus: setting organization %q and project %q at %s", d.EarthlyOrgName, d.EarthlyProjectName, time.Now().Format(time.RFC3339Nano))
 		analytics.AddEarthfileProject(d.EarthlyOrgName, d.EarthlyProjectName)
-		if a.cli.Flags().Logstream {
-			a.cli.LogbusSetup().SetOrgAndProject(d.EarthlyOrgName, d.EarthlyProjectName)
-			if doLogstreamUpload {
-				a.cli.LogbusSetup().StartLogStreamer(cliCtx.Context, cloudClient)
-			}
+		if !a.cli.Flags().Logstream {
+			return nil
+		}
+		setup := a.cli.LogbusSetup()
+		setup.SetOrgAndProject(d.EarthlyOrgName, d.EarthlyProjectName)
+		setup.SetGitAuthor(gitCommitAuthor, gitConfigEmail)
+		_, isCI := analytics.DetectCI(a.cli.Flags().EarthlyCIRunner)
```

Comment:
nit: not urgent, but might be good to move the `DetectCI` function to a neutral package.

Reply:
Good point. It's not necessarily tied to analytics.